client/app/project/project.controller.js

```diff
@@ -7,37 +7,98 @@
      */
     angular
         .module('taskingManager')
-        .controller('projectController', ['$location', 'mapService', 'projectService', 'styleService', projectController]);
+        .controller('projectController', ['$scope', '$location', 'mapService', 'projectService', 'styleService', projectController]);
 
-    function projectController($location, mapService, projectService, styleService) {
+    function projectController($scope, $location, mapService, projectService, styleService) {
         var vm = this;
         vm.project = null;
         vm.map = null;
         vm.currentTab = '';
+        vm.mappingStep = '';
+
+        //selected task
+        vm.selectedTask = null;
+        vm.isSelectTaskMappable = false;
+
+        //interaction
+        var select = new ol.interaction.Select({
+            style: styleService.getSelectedStyleFunction
+        });
 
         activate();
 
         function activate() {
             //TODO: Set up sidebar tabs
             vm.currentTab = 'description';
+            vm.mappingStep = 'select';
             mapService.createOSMMap('map');
             vm.map = mapService.getOSMMap();
+
+            vm.map.addInteraction(select);
+            select.on('select', function (event) {
+                $scope.$apply(function () {
+                    var feature = event.selected[0];
+                    onTaskSelection(feature);
+                });
+            });
+
             var id = $location.search().project;
             initialiseProject(id);
-            //TODO: put the project metadata (description instructions on disebar tabs
+            //TODO: put the project metadata (description instructions on siedbar tabs
         }
 
+        vm.selectRandomTask = function () {
+            var task = getRandomMappableTask(vm.project.tasks);
+            if (task) {
+                //iterate layers to find task layer
+                var layers = vm.map.getLayers();
+                for (var i = 0; i < layers.getLength(); i++) {
+                    if (layers.item(i).get('name') === 'tasks') {
+                        var feature = layers.item(i).getSource().getFeatures().filter(function (feature) {
```

Comment:
Instead of iterating over the layers, you could store the vector layer as a controller variable and reference it directly.